src/list/whole/component.rs

```diff
@@ -51,6 +51,10 @@ pub struct Model<T> {
     pub(super) sort_list_kind: Rc<RefCell<Option<SortListKind>>>,
 
     phantom: PhantomData<T>,
+
+    pub(super) modal_index: Option<usize>,
+    pub(super) modal_content: Vec<String>,
+    pub(super) title_id: String,
```

Comment:
Since only one modal is allowed at a time, I think we need only the selected `modal_content` and `title_id`. If I am right, these three can be combined into just one:

```rust
    pub(super) modal: Option<(String, String)>, // (String, String) = (content,  title_id)
```

Reply:
This order of (title, content) would be better.

Reply:
Done